### PR TITLE
deflake TestStaticPodFallbackConditionController

### PR DESCRIPTION
**Pull request by aojea**
The order of pods yields different results, however, the test only
look for one possible outcome.
Instead of complicating the test to deal with all the different
possible returns, we can use a custom podlister that returns
the pods in a deterministic order.

Signed-off-by: Antonio Ojea <aojea@redhat.com>

Tested with 
```
go test -v ./pkg/operator/staticpod/controller/staticpodfallback -run TestStaticPodFallbackConditionController -count 1 -c
$ stress ./staticpodfallback.test 
5s: 4406 runs so far, 0 failures
10s: 8767 runs so far, 0 failures
```

### CODE REVIEW

File: pkg/operator/staticpod/controller/staticpodfallback/static_pod_fallback_condition_test.go [hunk 1]
```diff
@@ -173,3 +175,38 @@ func newPod(phase corev1.PodPhase, ready corev1.ConditionStatus, revision, name
 
 	return &pod
 }
+
+// we need a deterministic pod lister for tests
+// otherwise the order of the pods will yield different results
+type orderedPodNamespaceLister struct {
+	podLister corev1listers.PodNamespaceLister
```
Comment (sttts) [inline pkg/operator/staticpod/controller/staticpodfallback/static_pod_fallback_condition_test.go:182]: why not deeply nest without name? Then you don't need the Get func

**Comment by sttts** [comment]
/lgtm
/approve

This not only applies to the unit test, but potentially also stops condition flapping in a real cluster.

File: pkg/operator/staticpod/controller/staticpodfallback/static_pod_fallback_condition_test.go (hunk 1 above)
Comment (aojea) [reply pkg/operator/staticpod/controller/staticpodfallback/static_pod_fallback_condition_test.go:182]: First interface ive seen :)

**Comment by aojea** [comment]
> This not only applies to the unit test, but potentially also stops condition flapping in a real cluster.

Should I add it to the real controller too?

**Pull request merged**